### PR TITLE
add config reader and yaml file

### PR DESCRIPTION
**Pull request by SmirnovGrigory**

### CODE REVIEW

File: include/async_image_loader.h [hunk 1]
```diff
@@ -17,7 +17,7 @@ using moodycamel::ReaderWriterQueue;
 
 class AsyncImageLoader {
 public:
-    explicit AsyncImageLoader(const char* folder, size_t start_frame,
+    explicit AsyncImageLoader(const std::string folder, size_t start_frame,
```
Comment (ikryukov) [inline include/async_image_loader.h]: const char* is better or use reference to get rid of local copy

File: include/config_reader.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
+//
+// Created by Smirnov Grigory on 22.01.2022.
+//
+
+#ifndef CONFIG_READER_H
```
Comment (ikryukov) [inline include/config_reader.h]: #pragma once

File: include/config_reader.h [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    std::string path;
+    double focal;
+    double cx;
+    double cy;
+    double bf;
+    int start_frame;
+    int end_frame;
+    bool show_gt;
+public:
+    ConfigReader(std::string filename) {
```
Comment (ikryukov) [inline include/config_reader.h]: const string&

File: include/config_reader.h [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+#ifndef CONFIG_READER_H
+#define CONFIG_READER_H
+
+#include <string>
+#include <iostream>
+#include <exception>
+#include <opencv2/core.hpp>
+
+class ConfigReader {
```
Comment (ikryukov) [inline include/config_reader.h]: if it is reader, then it should read config (fill structure with fields) from file and returns it, no need to store values in reader.

File: src/main.cpp [hunk 5]
```diff
@@ -16,12 +17,12 @@
 int main(int argc, char** argv)
 {
     int init_frame_id = 0;
-
+    ConfigReader reader("F:\\repos\\svo\\config.yaml");
```
Comment (ikryukov) [inline src/main.cpp]: use relative paths: "./config.yaml"

**Review by ikryukov — CHANGES_REQUESTED** [review]

File: include/config_reader.h (hunk 4 above)
Comment (SmirnovGrigory) [reply include/config_reader.h]: ok.done

File: include/config_reader.h (hunk 3 above)
Comment (SmirnovGrigory) [reply include/config_reader.h]: done

File: include/config_reader.h (hunk 2 above)
Comment (SmirnovGrigory) [reply include/config_reader.h]: done

File: src/main.cpp (hunk 5 above)
Comment (SmirnovGrigory) [reply src/main.cpp]: It doest work if i use relative path to config((

Comment (SmirnovGrigory) [reply src/main.cpp]: fixed with ../../config.yaml

**Review by ikryukov — APPROVED** [review]

**Pull request merged**